irods_capability_automated_ingest/test/test_s3_bucket_scan.py

```diff
@@ -409,6 +409,31 @@ def test_s3_with_register_as_replica_sync(self):
             finally:
                 self.minio_client.remove_object(self.bucket_name, new_object_name)
 
+    def test_register_to_deep_nonexistent_subcollection_does_not_hang_forever__issue_124(
+        self,
+    ):
+        operation = Operation.REGISTER_SYNC
+        event_handler_contents = test_s3_sync_operations.get_event_handler(operation)
+        # If the destination collection's parent collection does not exist, the bug is that the sync job will hang
+        # forever.
```

Comment:
This comment should probably read like this:

```suggestion
        # The destination collection needs to have enough path elements to exceed the number of path elements in
        # the "path" to the S3 object.
```

Reply:
Done